encoding/ssz/query/tag_parser.go

```diff
@@ -0,0 +1,114 @@
+package query
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+const (
+	// sszMaxTag specifies the maximum capacity of a variable-sized collection, like an SSZ List.
+	sszMaxTag = "ssz-max"
+
+	// sszSizeTag specifies the length of a fixed-sized collection, like an SSZ Vector.
+	// A wildcard ('?') indicates that the dimension is variable-sized (a List).
+	sszSizeTag = "ssz-size"
+)
+
+// SSZDimension holds parsed SSZ tag information for current dimension.
+//
+// NOTE: This struct stores raw string without parsing to uint64,
+// as it might have a wildcard ('?') indicating variable size.
+type SSZDimension struct {
+	SizeValue string // Current dimension for ssz-size
+	MaxValue  string // Current dimension for ssz-max
+	HasSize   bool
+	HasMax    bool
+}
+
+// ParseSSZTag parses SSZ-specific tags (like `ssz-max` and `ssz-size`)
+// and returns the first dimension and remaining tag.
```

Comment:
```suggestion
// and returns the first dimension and the remaining SSZ tags.
```
At first I thought the function returns the full remaining JSON tag